### PR TITLE
updated update schemas

### PR DESCRIPTION
**Pull request by jjidt**
- allows empty field values on creation
- breaks out create and update schemas using helpers
- partialFields schema helper
- updates documentation

NOTE: the previous swagger viewer I suggested wasn't showing payload examples, use `OpenAPI Preview` extension instead

### CODE REVIEW

**Review by jcontonio — DISMISSED** [review]
* I can update fields with just updating a name, or just updating a value 👍 
* I can create a field with just a name and a type (string or number) 👍 
* I get "name" is required if I try to create a field with just a value, that seems good, although I see details response is an array and I am only getting one validation error at a time. Any way to get all the errors?

Documentation updates:

* StringField, TextField, and NumberField all have value as required
* UpdateField only has `value`, but I'd expect `name` here as well so I can update the name
* Field only has `id` as required, expect `id` and `name` - name isn't listed in Field at all
* Should Document have `projectId` as required?

**Comment by jcontonio** [comment]
Oh, also `order` on UpdateField?

**Comment by jjidt** [comment]
update, I think the documentation correctly reflects the current state of the API

added order as a field return value, but we're not currently updating order, I'm thinking that will be it's own endpoint with side effects

the multi-validation error thing is probably a Joi setting, but we can handle that in the future

File: api/src/controllers/payloadSchemas/helpers.js [hunk 1]
```diff
@@ -13,25 +19,53 @@ const resourceToSchema = {
...
 }
 
+/**
+ * Use the baseSchema of a resource and limit the fields for validation
+ * be required
+ * @param {String} resource the resource name
+ * @param  {...String} fields field names found in schema to use in validation
+ * @return {Joi Schema} schema the compiled schema with included fields
+ */
+const partialFields = (resource, ...fields) => {
```
Comment (dheniges) [inline api/src/controllers/payloadSchemas/helpers.js:43]: ah nice

**Review by dheniges — APPROVED** [review]
Looks good to me

**Pull request merged**